### DISCUSSION TITLE
Add new CSP support: NCP-VPC

### DISCUSSION THREAD

**Discussion by jihoon-seo** [cloud-barista/cb-tumblebug#1115] [General]
입력해야 하는 Credential fields: 
https://github.com/cloud-barista/cb-spider/blob/master/api-runtime/rest-runtime/test/connect-config/14.ncpvpc-conn-config.sh 참고

```json
{
  "CredentialName": "ncpvpc-credential01",
  "ProviderName": "NCPVPC",
  "KeyValueInfoList": [
    {
      "Key": "ClientId",
      "Value": "XXXXXXXXXXXXXXXXXX"
    },
    {
      "Key": "ClientSecret",
      "Value": "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"
    }
  ]
}
```

NCP (Classic) 와 동일한 credential 사용

---

2022-05-16:
TB 테스트 스크립트 수정 없이 ./create-all.sh & ./clean-all.sh 실행으로
자원 및 VM 생성/삭제가 되는 것을 확인했습니다.
(관련 TB PR: #1114)

**Comment by jihoon-seo** [comment]
2022년 9월 29일 기준, 
CB-Spider 및 CB-Tumblebug 최신버전 (latest master/main) 에서 
CB-Tumblebug 테스트 스크립트 수정 없이
NCP-VPC Korea 1 (Region: `KR` // Zone: `KR-1`) 에 대해
VM 생성/삭제 까지 되는 것을 확인했습니다.
(NLB, DataDisk 는 CB-Spider NCP-VPC 드라이버에 아직 구현되지 않았습니다.)

- CB-Spider: v0.6.9 + a ([`c34717c`](https://github.com/cloud-barista/cb-spider/commit/c34717cd7e4af72e402262069aff0e5de8fa764b))
- CB-Tumblebug: v0.6.3 + a (3671158)

[참고]
9월 23일에 CB-Spider에서 NCP-VPC에 대해 Japan region & zones 정보를 추가해 주셨습니다. ([관련 PR](https://github.com/cloud-barista/cb-spider/pull/768))
이를 CB-Tumblebug 테스트 스크립트에도 반영했습니다.

**Comment by seokho-son** [comment]
Tested on 2022-10-04
./create-all.sh -f ../testSet-ncp.env -n ncp01
![image](https://user-images.githubusercontent.com/5966944/193755490-c3402710-f094-4d60-a511-592cca3b82a6.png)

./gen-sshKey.sh -f ../testSet-ncp.env -n ncp01
![image](https://user-images.githubusercontent.com/5966944/193755645-d8265c9b-1658-45a9-8902-4f8e108f3352.png)

**Comment by jihoon-seo** [comment]
[동일 MCIS, 동일 subGroup에 속하는 VM 2개 생성]
OK

[NLB 생성 시도]
OK

![image](https://user-images.githubusercontent.com/46767780/204457571-d5de5fef-fe88-4afc-a00e-21ac3a9c43f2.png)

![image](https://user-images.githubusercontent.com/46767780/204457637-edcfed0d-01fe-4370-b9f0-96daa0a9eba7.png)

[NLB에 연결된 VM 2개 중 하나 제거]
OK

![image](https://user-images.githubusercontent.com/46767780/204458148-fa4f1687-562c-4190-ab6d-0d50ddca02ce.png)

![image](https://user-images.githubusercontent.com/46767780/204458210-6c16035e-45f1-4dc0-b241-a877063416bb.png)

[제거했던 VM을 다시 추가]
OK

![image](https://user-images.githubusercontent.com/46767780/204458695-1dd8e8fb-8524-4fdc-a173-8cfd646fb7f8.png)

![image](https://user-images.githubusercontent.com/46767780/204458756-d16046bc-250e-4081-a8d7-824fe2cc20db.png)

[Health 확인]
OK

```
❯ ./health-NLB.sh -n jhseo -c ncpvpc -r 1
####################################################################
## 10. NLB: Get Health
####################################################################

{
  "AllVMs": [
    "ncpvpc-sgn-4-1",
    "ncpvpc-sgn-4-2"
  ],
  "HealthyVMs": [
    "ncpvpc-sgn-4-1",
    "ncpvpc-sgn-4-2"
  ],
  "UnHealthyVMs": null
}
```

[NLB 삭제]
OK
